File: Others/PG_프린터.js
// 프로그래머스 프린터
// 큐를 응용한 문제, js에서 인덱스와 값을 접근하는 방식을 잘 익히자!

function solution(priorities, location) {
    let paper = []
    for (let i in priorities) {
        paper.push([parseInt(i), priorities[i]])
    }
    let answer = []
    
    while (paper.length > 0) {
        let temp = []
        for (let p of paper) {
            temp.push(p[1])
        }
        let current = paper.shift()
        
        if (current[1] >= Math.max(...temp)) {
            answer.push(current[0])
        } else {
            paper.push(current)
        }
    }
    
    let ans = 1
    for (let a of answer) {
        if (a == location) {
            return ans
        }
        ans++
    }
}

File: Others/PG_프린터.py
# 프로그래머스
## 생각해보면 정말 단순한 문제
## 너무 어렵게만 생각하지 말자...

def solution(priorities, location):
    paper = [(p, i) for i, p in enumerate(priorities)]
    ans = []

    while paper:
        num, idx = paper.pop(0)
        temp = [p for p, i in paper]
        max_num = 0
        if temp:
            max_num = max(temp)

        if num >= max_num:
            ans.append(idx)
        else:
            paper.append((num, idx))

    for a in range(len(ans)):
        if ans[a] == location:
            return a + 1